src/tbp/monty/frameworks/environments/embodied_data.py

```diff
@@ -336,6 +347,16 @@ def change_object_by_idx(self, idx):
             "semantic_id": self.semantic_label_to_id[self.object_names[idx]],
             **self.object_params,
         }
+        if self.parent_to_child_mapping is not None:
+            if self.primary_target["object"] in self.parent_to_child_mapping:
+                self.consistent_child_objects = self.parent_to_child_mapping[
+                    self.primary_target["object"]
+                ]
+            else:
+                logger.warning(
+                    f"target object {self.primary_target['object']} not in",
+                    " parent_to_child_mapping",
+                )
```

Comment:
Does it make sense to run this code in the event that `self.parent_to_child_mapping` is an empty dictionary, i.e. `{}`? Currently it will be run in that case, but if it doesn't make sense (I could see it either way), it would make the conditional simpler to just do `if self.parent_to_child_mapping` and that would exclude both `None` and `{}`.